Cython/Utility/ObjectHandling.c

```diff
@@ -285,11 +285,6 @@ static CYTHON_INLINE PyObject *__Pyx_PyObject_GetItem(PyObject *obj, PyObject* k
 static PyObject *__Pyx_PyObject_GetIndex(PyObject *obj, PyObject* index) {
     PyObject *runerr;
     Py_ssize_t key_value;
-    PySequenceMethods *m = Py_TYPE(obj)->tp_as_sequence;
-    if (unlikely(!(m && m->sq_item))) {
-        PyErr_Format(PyExc_TypeError, "'%.200s' object is not subscriptable", Py_TYPE(obj)->tp_name);
-        return NULL;
-    }
```

Comment:
This change now means the function won't fail (but may crash), even if the sequence isn't subscriptable, since the check is elsewhere. I think this method is only used from `__Pyx_PyObject_GetItem` so it shouldn't break anything. But maybe it atleast requires a comment saying "obj MUST have tp_as_sequence" ?

Reply:
This comment actually looks like it isn't true. It looks like `__Pyx_GetItemInt_Fast` actually handles non-`tp_as_sequence` cases, and even a generic fallback `PyObject_GetItem` in `__Pyx_GetItemInt_Generic` at the end. Presumably the check for `tp_as_sequence` allows the compile to identify that only one code-path is ever used and inline only the relevant bits. So it may be necessary to produce optimized code, but probably doesn't crash.

Reply:
Added a comment now that I've put the function back. It does look like it won't crash even if were called on a non-sequence type but there is indeed no other caller at the moment so not sure how much it matters in either case.

Reply:
I'd still be slightly worried that removing `if (m && m->sq_item)` means that the whole of `__Pyx_GetItemInt_Fast` gets inlined and not just smaller one path from it. I'm not certain though so definitely don't change anything because I say so. @scoder this might be worth a closer look?

Reply:
The code in `__Pyx_GetItemInt_Fast` tries `mp_subscript` first, so I don't think it makes much of a difference. (Just guessing, but I don't feel like digging deeper here.)